### ISSUE TITLE
User cannot read password file

### ISSUE THREAD

**Issue by ryanlovett** [cloudpipe/keymaster#8]
With the caveat that I'm very new to docker...

Inside the container, /certificates is owned by the uid of the user who created $(pwd)/certificates on the host. When the container runs with the "ca" parameter, uid=1000 (hagrid) expects to be able to have access to /certificates/password. Since the file's owner on the host is not uid 1000, "${KEYMASTER} ca" fails:

host$ $ docker -v
Docker version 1.6.0, build 4749651
host$ ${KEYMASTER} ca
cat: /certificates/password: Permission denied
host$ docker run -it --rm -v $(pwd)/certificates/:/certificates/ cloudpipe/keymaster /bin/bash
hagrid@string:~$ ls -l /certificates/password
-rw------- 1 2354 2354 175 May  7 00:08 password
hagrid@string:~$ id
uid=1000(hagrid) gid=1000(hagrid) groups=1000(hagrid)

**Comment by gngdb** [comment]
Immediately hit this exact problem. Guessing the fix is to make a derivative image where the permissions are fixed?

**Comment by ramdhanyk** [comment]
Hi @ryanlovett  / @gngdb,

I am facing the same issue. Were you able to resolve this issue? If you did, any guidance is appreciated.

**Comment by ryanlovett** [comment]
Yeah, I ended up forking at https://github.com/ryanlovett/keymaster.

**Comment by ramdhanyk** [comment]
Thanks for the message. I will use your fork. You should probably send a pull request.

**Comment by rgbkrk** [comment]
If you want to roll with a PR @ryanlovett, I'd be happy to add you as a maintainer on github.com/cloudpipe/keymaster as well as give you access to the Docker Hub image.

**Comment by rgbkrk** [comment]
Our reasoning on using a particular non-root user was before we had real Linux namespaces (as will be in coming releases of Docker). I'm comfortable with your changes @ryanlovett.

**Comment by ryanlovett** [comment]
Thanks for reviewing the changes. Before anything gets pushed to cloudpipe, I think I need to check with @jhamrick to make sure my changes don't mess things up for her. I think this comment will ping her?

**Comment by ramdhanyk** [comment]
@ryanlovett 

I am really new to docker. So, this question might be amateurish. 

When I try to run the ${KEYMASTER} mkpassword, I am getting the error that mkpassword is not in the $PATH.

This made me wonder, whether I should run any script once I pull the git repo. If not, should I just go ahead and export the ~/keymaster/bin directory and run the ${KEYMASTER} mkpassword

Thanks for your input.

**Comment by jhamrick** [comment]
Looks fine to me! I don't think it will cause any problems on my end.

**Comment by ryanlovett** [comment]
@ramdhanyk

The instructions below should work, but let me know if there is a a
problem. Make sure to start with a clean git clone.

git clone https://github.com/ryanlovett/keymaster
cd keymaster/
docker build -t somename .
mkdir certificates
KEYMASTER="docker run --rm -v $(pwd)/certificates/:/certificates/ somename"
${KEYMASTER} mkpassword
${KEYMASTER} ca
${KEYMASTER} signed-keypair -n service1 -h service1.host.com
...

Ryan

**Comment by ramdhanyk** [comment]
Ryan,

Thanks for the message. I was able to get it to work. Thanks a lot for your help.
